tests/framework/prototype.py

```diff
@@ -154,6 +163,19 @@ def check_rule_sets(self):
         """When the system converges, check all registered RuleSets."""
         for ruleset in self.rulesets:
             ruleset.check_rule_set(self.custom_resource_object, self.namespace)
+    def clean_up(self):
+        """Cleanup the CR."""
+        raise NotImplementedError
+
+    def event_timeout(self, fail_event, expected_head_pods, expected_worker_pods):
```

Comment:
* Move the log functions to `wait`.
```python
logger.info("%s failed to converge in %d seconds.", fail_event, self.timeout)
logger.info("expected_head_pods: %d, expected_worker_pods: %d",
    expected_head_pods, expected_worker_pods)
```

* Define a function for cluster information (e.g. Pod & Service).